core/src/main/java/org/apache/calcite/sql/validate/SqlUserDefinedFunction.java

```diff
@@ -61,9 +64,9 @@ public SqlUserDefinedFunction(SqlIdentifier opName, SqlKind kind,
       SqlReturnTypeInference returnTypeInference,
       SqlOperandTypeInference operandTypeInference,
       @Nullable SqlOperandMetadata operandMetadata,
-      Function function) {
+      Function function, SqlSyntax syntax) {
```

Comment:
you cannot change public constructors - this is a breaking change for clients.
you have to keep this constructor, and add another one.

Reply:
Hi @mihaibudiu thank you for your suggestion. I found this public constructor, which is only called in the `CalciteCatalogReader#toOp` method. When we add SqlSyntax to the toOp method, the original constructor that does not contain SqlSyntax will not be called.

In this case, does it make sense to keep a constructor that is not called? In addition, I don't know whether this constructor will be called by the user. If the user can call it, then we should indeed keep the original constructor.

Reply:
Since Calcite is an open-source library you cannot assume that the only code that calls these functions is in the Calcite tree. There are lots of third party projects using Calcite, and any of them may be calling any of the public functions. You cannot break compatibility unless you change the major version of the project.

Reply:
Thanks for your suggestion, I will keep the original constructor and add a new one for passing SqlSyntax.